sockeye/model.py

```diff
@@ -223,7 +226,7 @@ def _get_embed_weights(self, prefix: str) -> Tuple[mx.sym.Symbol, mx.sym.Symbol,
                                                 self.config.config_embed_target.num_embed))
 
         w_out_target = mx.sym.Variable(prefix + "target_output_weight", dtype='float32',
-                                       shape=(self.config.vocab_target_size, self.decoder.get_num_hidden()))
+                                       shape=(self.config.vocab_target_size - self.config.num_pointers, self.decoder.get_num_hidden()))
```

Comment:
I realized that the approach will fail if weight tying between the output projection and embedding matrices is turned on. This is because the variable is overwritten below. However, you can not simply change the shape because you do want to learn embeddings for `<ptr*>` symbols. One solution would be to slice the embedding weight below when weight tying is used together with the pointers. What do you think?

Reply:
Yes, I implemented your suggestion of slicing the output projection weights when weight tying is used together with the pointers.